Docs: Remove warning about path with spaces on Windows

timonsku opened the pull request:
Works just fine if I point it to the default install path from the Arm GCC installer which contains several spaces.
"C:\Program Files (x86)\GNU Arm Embedded Toolchain\x.x.x\"

tejlmand commented:
@mbolivar-nordic you originally added this commit https://github.com/zephyrproject-rtos/zephyr/commit/0d811b9aee6a6410f73fdfb4d4072d4d16b05a52 with this sentence:
>   Like the Zephyr repository, do not install the toolchain in a directory with spaces anywhere in the path.

while it is correct you should not install Zephyr at a location containing spaces (one reason being the fact we treat space as value separator for `CONF_FILE` and similar), then I'm unsure of the reason why the toolchain cannot be installed in path containing spaces.

Do you remember why this was added ?

mbolivar-nordic commented:
> Do you remember why this was added ?

I don't remember anymore, sorry. If it's wrong, let's just remove it.

tejlmand reviewed (dismissed):
lgtm.

Please read the contribution guidelines:
https://docs.zephyrproject.org/latest/contribute/#commit-guidelines

timonsku commented:
Ah sorry, missed the guidelines. Will remember for next time.

mbolivar-nordic commented:
> Will remember for next time.

This needs to be fixed or the PR is not mergeable

mbolivar-nordic commented:
You need to:

1. Fix your signed-off-by line to use a legal name and a real email (https://docs.zephyrproject.org/latest/contribute/index.html#dco-sign-off-methods)
2. please use `doc:` instead of `Docs:`

mbolivar-nordic reviewed (changes requested):
My mistake, meant to -1 not +1

tejlmand reviewed (dismissed):
lgtm

carlescufi commented:
@timonsku could you please fix your commit message as per @mbolivar-nordic's instrucitons?

timonsku committed:
doc: Add a notice that you are free to choose compiler path

Works just fine if I point it to the default install path from
the Arm GCC installer which contains several spaces.
"C:\Program Files (x86)\GNU Arm Embedded Toolchain\x.x.x\"

The previous notice about not using spaces made sense to have this
very specific path but this might confuse people who install
ARM GCC with the provided installer which puts it into a path
under C:\Program FIles, to avoid people moving their installation
and possibly breaking a different setup I thinks its good
to make that clear that this is not a hard coded assumption
made across Zephyr and is just a matter of an ENV variable.

Signed-off-by: Timon Skerutsch <timon@diodes-delight.com>
Co-Authored-By: Stephanos Ioannidis <root@stephanos.io>

timonsku commented:
Sorry for the delay! Had some busy weeks.
Thank for bearing with me. I squashed and amended the commits as instructed.
I hope that now follows the guidelines correctly.

mbolivar-nordic commented:
@timonsku no worries! You are almost there. Just please change this:

```
Signed-off-by: Timon Skerutsch timon@diodes-delight.com
```

To this:

```
Signed-off-by: Timon Skerutsch <timon@diodes-delight.com>
```

and all will be well.

timonsku commented:
ah yes, fixed!

stephanosio reviewed (changes requested):
```
1: T6 Title has leading whitespace: " doc: Add a notice that you are free to choose compiler path"
3: UC4 Line exceeds max length (115>75): "Works just fine if I point it to the default install path from the Arm GCC installer which contains several spaces."
6: UC4 Line exceeds max length (1[27](https://github.com/zephyrproject-rtos/zephyr/runs/5699365541?check_suite_focus=true#step:9:27)>75): "The previous notice about not using spaces made sense to have this very specific path but this might confuse people who install"
7: UC4 Line exceeds max length (127>75): "ARM GCC with the provided installer which puts it into a path under C:\Program FIles, to avoid people moving their installation"
8: UC4 Line exceeds max length (117>75): "and possibly breaking a different setup I thinks its good to make that clear that this is not a hard coded assumption"
```

timonsku commented:
fixed

stephanosio commented on doc/getting_started/toolchain_3rd_party_x_compilers.rst:
```diff
@@ -11,17 +11,15 @@ external organization. Several of these are available.
-      :file:`C:\\gnu_arm_embedded`.
+      On Windows, we'll assume for this guide that you install into the directory
+      :file:`C:\\gnu_arm_embedded`. You can also choose the default installation
+      path used by the ARM GCC installer, you will need to adjust the path
```
```suggestion
      path used by the ARM GCC installer, in which case you will need to adjust the path
```

timonsku replied:
sure, lgtm

stephanosio reviewed (commented):
It looks like you have added a new commit while addressing the review comment.

Please squash that commit into the first one; also, there is no need to add `Co-authored-by: Stephanos ...` for minor review comments like this :)

timonsku commented:
oh sorry I thought I had already fixed this. Commit is squashed.

stephanosio commented:
> oh sorry I thought I had already fixed this. Commit is squashed.

@timonsku Thanks. It looks like there is a merge conflict with the latest main. Could you rebase/resolve conflicts and push again?

timonsku commented:
Rebased to latest upstream main.
I just noticed it auto added the co-authored to the commit message, I hope thats ok?

stephanosio commented:
> I just noticed it auto added the co-authored to the commit message, I hope thats ok?

yea, no worries :)
